### ISSUE TITLE
Autocorrect and autocapitalization not working on iOS Safari

### ISSUE THREAD

**Issue by josephdpurcell** [0xsuk/agitcms#30]
# Description

The autocorrect and autocapitalization features do not work on iOS for the Markdown input.

# Steps to reproduce

1. Install Agit CMS
2. Login
3. Create a file
4. In the content area no characters get autocapitalized (eg after a period)
5. Notice that entering two spaces does not result in a period followed by a space
6. Notice that misspelled words do not result in a prompt for the correct spelling

# Versions

agitcms version: 2.0.5
node version: 16.18.0
iOS Safari version: 15

# Recording

I am attaching a recording of the behavior.

https://user-images.githubusercontent.com/778111/209684206-43ac2fd2-9c5c-4f2c-adb5-884493703e36.MOV

**Comment by josephdpurcell** [comment]
Some context: I am in search of a "good" markdown editing experience, which in my opinion includes these features. I am taking notes on https://github.com/directus/directus/discussions/16762, which includes a link to a spreadsheet that has an evaluation of various editors and CMS's.

The lack of support for autocorrect and autocapitalization is a common issue I'm finding. Here are some related tickets:

* Ghost CMS: https://github.com/TryGhost/Ghost/issues/16049
* Pagekit: https://github.com/pagekit/pagekit/issues/975
* Directus: https://github.com/directus/directus/issues/16699

**Comment by 0xsuk** [comment]
Issue confirmed.
This is probably an issue with Codemirror, a dependency that powers the markdown editor.
The reason I can tell so is that all of the repo you listed above use Codemirror internally.
(e.g. https://github.com/directus/directus/search?q=codemirror)

I found a [similar issue](https://github.com/codemirror/dev/issues/433) in Codemirror.  
You might want to consider contributing to Codemirror or search for CMS that does not use Codemirror.

**Comment by josephdpurcell** [comment]
Thanks for the prompt reply @0xsuk!

I confirmed that CodeMirror5 has the issue, see https://github.com/codemirror/codemirror5/issues/7009.

But CodeMirror6 does not, see https://github.com/codemirror/dev/issues/1020.

My adventure started with a simple diagnosis of the issue on Directus but has evolved into evaluating every editor and CMS I can, but time is limiting. Here is my current results: https://docs.google.com/spreadsheets/d/18v2G44v0ix41xbAseOIcMcFOMNluc67IJz9f2U5IpRo/edit?usp=sharing

**Comment by 0xsuk** [comment]
https://github.com/0xsuk/agitcms/commit/30beb0f5b598dcd78cdfe63e62e9056472a6fc80 have fixed the issue.

I don't know why but there's no squiggly red warning showing up under the incorrect word. I think that's not much of a problem so I'm closing this issue.

**Comment by josephdpurcell** [comment]
Yes! Amazing! I just confirmed it works using the same criteria for which I reported.

Thank you for making this change so quickly!

Of 16 CMS's I evaluated, at present only Agit CMS and Typemill support the autocorrect/punctuate/capitalize features on iOS.

One additional change to consider would be to enable autocorrect. Based on my testing with CodeMirror6 this would not adversely effect the authoring experience of Markdown. (I may attempt to contribute said change if I can, but at minimum not likely for another week.)

PS. This is a really amazing project. There are several reasons why I like it more than others I've evaluated; I'll hopefully share more thoughts than just the spreadsheet at a future date.
